app/assets/javascripts/media_player_wrapper/avalon_player_new.es6

```diff
@@ -264,66 +342,98 @@ class MEJSPlayer {
    */
   handleSuccess (mediaElement, originalNode, instance) {
     this.mediaElement = mediaElement
+
     // Show the player in success callback
     this.revealPlayer(instance)
+
     // Grab instance of player
     if (!this.player) {
-      this.player = mediaElement
+      this.player = this.mediaElement
     }
-    mediaElement.addEventListener('canplay', this.handleCanPlay.bind(this))
+
+    // Handle 'canplay' events fired by player
+    this.mediaElement.addEventListener('canplay', this.handleCanPlay.bind(this))
+
     // Show highlighted time in time rail
     if (this.highlightRail) {
-      this.highlightTimeRail.apply(this)
+      // Create our custom time rail highlighter element
+      this.createTimeHighlightEl()
+      this.highlightTimeRail.apply(this, [this.activeSegmentId])
     }
+
+    // Listen for timeupdate events in player, to show / hide highlighted sections, etc.
+    this.mediaElement.addEventListener('timeupdate', this.handleTimeUpdate.bind(this))
   }
 
   /**
-   * Highlight a section of the Mediaelement player's time rail
-   * @function highlightTimeRail
-   * @return {void}
+   * Callback function to handle MEJS's 'timeupdate' event
+   * Essentially this is a polling function to show/display a highlighted segment region
+   * in MEJS's UI time rail
+   * @function handleTimeUpdate
+   * @return {[type]} [description]
    */
-  highlightTimeRail () {
-    let highlightSpanEl = document.getElementById('content').querySelector('.mejs-highlight-clip')
-    let styleAttribute = ''
+  handleTimeUpdate () {
+    const segmentsMap = this.segmentsMap
+    const currentTime = this.player.getCurrentTime()
+    let activeId = ''
+
+    // Get the segment for player's current time value (if one exists)
+    for (let segmentId in segmentsMap) {
+      let begin = parseFloat(segmentsMap[segmentId].fragmentbegin)
+      let end = parseFloat(segmentsMap[segmentId].fragmentend)
+
+      if ( (segmentsMap.hasOwnProperty(segmentId)) && (currentTime >= begin && currentTime < end) ) {
+        activeId = segmentId
```

Comment:
Maybe break the for-loop here? No need to continue through the structure. Side effect is that for items with structure that overlaps, it will pick the first match instead of the last.

Reply:
Totally, another good catch.

Reply:
Actually, this is a tricky one.  With the `break`, it picks the first, however in an overlapped test item, it wrongly picks the first if a later, overlapping item is clicked upon.

Reply:
Here's the structured metadata I'm testing against.  I could be missing something, but for now this will only work with the break in there.  Feel free to pull down and let me know if we're envisioning different test data?

```
<?xml version="1.0"?>
<Item label="Bunny Video">
    <Div label="Funny Parts">
        <Span label="Funny Part 1" begin="0" end="00:10"/>
        <Span label="Funny Part 2" begin="00:12" end="00:17"/>
    </Div>
    <Div label="Scary Parts">
        <Span label="Scary Part 1" begin="00:25" end="00:35"/>
        <Span label="Scary Part Overlap" begin="00:30" end="00:40"/>
        <Span label="Scary Part Overlap" begin="00:35" end="00:46"/>
        <Span label="Scary Part 2" begin="00:38" end="00:49"/>
    </Div>
</Item>
```

Reply:
I think we need to find out if overlapping is even legal. I suspect it shouldn't be, because expected behavior is not well-defined.